group_project_v2/notifications.py

```diff
@@ -144,39 +144,41 @@ def on_before_studio_delete(self, _course_id, services):
             for timer_suffix in (NotificationTimers.OPEN, NotificationTimers.DUE, NotificationTimers.COMING_DUE):
                 notifications_service.cancel_timed_notification(self._get_stage_timer_name(timer_suffix))
 
-
-class ActivityNotificationsMixin(object):
-    # While we *should* send notification, if there is some error here, we don't want to blow the whole thing up.
     @log_and_suppress_exceptions
     def fire_file_upload_notification(self, notifications_service):
+
+        log.info('{}.fire_file_upload_notification on location = {}'.format(self.__class__.__name__, self.location))
+
         # this NotificationType is registered in the list of default Open edX Notifications
         msg_type = notifications_service.get_notification_type(NotificationMessageTypes.FILE_UPLOADED)
 
         workgroup_user_ids = []
         uploader_username = ''
-        for user in self.workgroup.users:
+        for user in self.activity.workgroup.users:
```

Comment:
@ziafazal nit: `workgroup` should be available at `self` - it is provided by `WorkgroupAwareXBlockMixin` which is mixed both to Activity and Stage XBlocks.